database/migrations/2023_10_05_193906_create_groups_table.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class() extends Migration {
+    public function up(): void
+    {
+        Schema::create("groups", function (Blueprint $table): void {
+            $table->ulid("id")->primary();
+            $table->string("name");
+            $table->string("form");
+            $table->string("course_semester_id")->nullable();
+            $table->foreign("course_semester_id")->references("id")->on("course_semester")->onDelete("set null");
```

Comment:
```suggestion
            $table->foreign("course_semester_id")->references("id")->on("course_semester")->onDelete("SET NULL");
```

As we discussed recently.